### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by teopiaz**
I decided to start documenting part of the source code. It might help newcomers (like me) to get through the code.
I would like to use doxygen documentation in order to, in the future, generate the usual html docs.

Will it makes sense?

When I will have a better understanding on the engine I could make some architectural description.

### CODE REVIEW

File: src/libgv/strcode.c [hunk 1]
```diff
@@ -1,5 +1,11 @@
 #include <SYS/TYPES.H>
 
+/**
+ * @brief Generate a simple hash code from a string.
+ *
+ * @param string input string
+ * @return int 32-bit hash
```
Comment (GirianSeed) [inline src/libgv/strcode.c:7]: GV_StrCode returns a 32-bit integer but the hash itself is only 16-bits. (Same thing with the MGS2 version where the hash is masked to 24-bits.)

File: src/Weapon/rcm.c [hunk 2]
```diff
@@ -78,11 +90,16 @@ void rcm_act_helper_80066B58(RcmWork *work, int flags)
     pPrim->line_g2.b0 = curRgb;
 }
 
+/**
+ * @brief Handles the behavior of the Nikita missile.
```
Comment (teopiaz) [inline src/Weapon/rcm.c:94]: This is wrong, is the Nikita launcher not the missile. The missile is a separated actor.
I am gonna fix in the whole file

File: src/Equip/gglsight.h [hunk 3]
```diff
@@ -8,6 +8,9 @@
 #include <LIBGTE.H>
 #include <LIBGPU.H>
 
+#define SCN_MASK_NIGHT_VISION 5
```
Comment (KieronJ) [inline src/Equip/gglsight.h:11]: These have defines already as ITEM_N_V_G & ITEM_THERM_G

**Comment by giuse94** [comment]
It would be nice to rename the documented functions as well. For example, `gglsight_act_helper_80077F70` could be `drawScanAndModeText_80077F70`.

**Comment by KieronJ** [comment]
Looks good, thanks for sending.

**Pull request merged**